### ISSUE TITLE
Footnotes in regular tblr

### ISSUE THREAD

**Issue by niveK77pur** [lvjr/tabularray#420]
As mentioned in #336 one can create footnotes using a combination of `note` and `\TblrNote`. However, when using the "default" tblr table, this does not work. Also, the examples in the documentation only illustrate this for tall and long tables. 

Would it be handy and possible to allow these footnotes for all types of tables? I have a very short table where I want to put a footnote for additional context, but the text is not strictly relevant to the table's content so I want to avoid cluttering it with the help of a footnote.

*Note that I am not asking for* page footnotes *as in the linked issue #336. I merely want the ability to create footnotes like is already possible for long and tall tables.*

# Example

In this example, the footnote does not appear anywhere. The footnote also does not appear when setting `baseline` or `expand` outer specifications (sec 3.2, table 3.2 in the documentation).

```latex
\documentclass{scrartcl}
\usepackage{tabularray}
\usepackage{blindtext}

\begin{document}

\blindtext

\begin{table}[ht]
  \caption{Normal table}
  \centering
  \begin{tblr}[
      %baseline,
      note{a} = {Footnote in a normal table},
    ]{
      colspec = {cc},
    }
    Key             & Value \\
    abc             & 1     \\
    def\TblrNote{a} & 2     \\
    ghi             & 3     \\
  \end{tblr}
\end{table}

\blindtext

\end{document}
```

![image](https://github.com/lvjr/tabularray/assets/10981161/f5848c5d-11af-4e7f-8999-33b7d728e0f9)

# Workaround

I can *fool* the package into putting a footnote by using `tall` or `long` for the table. I want the table to remain floating however, which is why I still keep it inside the table environment. However, the spacing above and below the table has increased, and I cannot quite figure out how to reduce it (I was looking at sec. 4.2.1, table 4.4 in the documentation)

```latex
\documentclass{scrartcl}
\usepackage{tabularray}
\usepackage{blindtext}

\begin{document}

\blindtext

\begin{table}[ht]
  \begin{tblr}[
      long,
      caption = {Long table},
      note{a} = {Footnote in a long table},
    ]{
      colspec = {cc},
    }
    Key             & Value \\
    abc             & 1     \\
    def\TblrNote{a} & 2     \\
    ghi             & 3     \\
  \end{tblr}
\end{table}

\blindtext

\end{document}
```

![image](https://github.com/lvjr/tabularray/assets/10981161/46cc5c87-4507-4739-9c9c-4c3d2adf3d75)

**Comment by muzimuzhi** [comment]
> However, when using the "default" tblr table, this does not work. Also, the examples in the documentation only illustrate this for tall and long tables.

"Default" tblr table has no head and foot components (they are ignores even if specified). You need "tall" tblr here. "Tall" table works just like "default" tblr table but typesets table head and foot components. Footnotes are part of the table head.

> However, the spacing above and below the table has increased, and I cannot quite figure out how to reduce it (I was looking at sec. 4.2.1, table 4.4 in the documentation)

You used "long" tblr table, which is typeset in its own paragraph, surrounded by (extra) vertical spaces `presep` and `postsep`.

An example for "tall" tblr tables is given below.
- The first group of tables (above the perfectly overlapping hrules) show that "tall" table takes just the same vertical space as "default" one.
- The third table (below the hrules) shows a "tall" table with footnote. It's caption is still specified using `\caption` command (the tblr caption is disabled by `label=none`).

<details><summary>Full example</summary>
<p>

```tex
\documentclass[twocolumn]{article}
\usepackage{tabularray}
\usepackage{lipsum}

\begin{document}

\lipsum[1][1-3]

\begin{table}[ht]
  \centering
  \caption{Normal table}
  \begin{tblr}{
      colspec = {cc},
    }
    Key             & Value \\
    abc             & 1     \\
    def             & 2     \\
    ghi             & 3     \\
  \end{tblr}
\end{table}

\lipsum[1][1-3]

\leavevmode\rule{\linewidth}{.4pt}\par

\newpage

\lipsum[1][1-3]

\begin{table}[ht]
  \centering
  \caption{Tall table}
  \begin{tblr}[
      tall,
      label=none,
    ]{
      colspec = {cc},
    }
    Key             & Value \\
    abc             & 1     \\
    def             & 2     \\
    ghi             & 3     \\
  \end{tblr}
\end{table}

\lipsum[1][1-3]

\leavevmode\clap{\rule{2\linewidth}{.4pt}}\par

\lipsum[1][1-3]

\begin{table}[ht]
  \centering
  \caption{Tall table}
  \begin{tblr}[
      tall,
      label=none,
      note{a} = {Footnote in a tall table},
    ]{
      colspec = {cc},
    }
    Key             & Value \\
    abc             & 1     \\
    def\TblrNote{a} & 2     \\
    ghi             & 3     \\
  \end{tblr}
\end{table}

\lipsum[1][1-3]

\end{document}
```

</p>
</details>

![image](https://github.com/lvjr/tabularray/assets/6376638/a97d1c1f-d6dc-48ae-94ff-af2c830bae7a)

**Comment by niveK77pur** [comment]
Thank you very much for the detailed information! I did notice that there was a distinction being made between *tall* and *long* tables, but it was not clear to me what exactly the differences were; hence I did not try using the tall table in this case.

Also thanks a lot for providing such a thorough example, you definitely showed me something new and neat there with the `label=none` as well!

I shall close this issue as my initial concern has been resolved; thanks again!
